### PR TITLE
Build, package, ship to S3

### PR DESCRIPTION
**Pull request by durch**
@russellpierce, @shalgrim this is more of an FYI, its just building and uploading to S3 ATM , I'll merge tomorrow so it gets pulled into branches and it will than be used as part of `data-science-env` build

### CODE REVIEW

**Comment by durch** [comment]
retest this please

**Comment by durch** [comment]
retest this please

**Comment by durch** [comment]
retest this please

**Comment by durch** [comment]
retest this please

**Review by shalgrim — APPROVED** [review]
LGTM

File: ci.R [hunk 1]
```diff
@@ -0,0 +1,2 @@
+devtools::build()
+#devtools::test()
```
Comment (russellpierce) [inline ci.R:2]: @durch It is now safe to test redshiftTools.  Maybe restore?

File: tests.dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+WORKDIR /tmp
+
+RUN Rscript bootstrap.R
+
+ADD . /code
+WORKDIR /code
+
+#RUN Rscript ci.R
+
+RUN R CMD build .
```
Comment (russellpierce) [inline tests.dockerfile:16]: ... or is that happening here?

Comment (durch) [reply tests.dockerfile:16]: Nowhere actually, this is just building `tar.gz`. `R CMD check` is still failing on `zapieR`, `database.R`

Comment (russellpierce) [reply tests.dockerfile:16]: I know we talked about this a bit before, but I didn't really get it then either.  I think what you are saying is that after you build redshiftTools then zapieR's CMD CHECK fails, but that if you zapieR's CMD CHECK alone (or with an older version of redshiftTools) it passes?

Comment (durch) [reply tests.dockerfile:16]: Actually it seems that as part of the `redshiftTools` CMD check the dependancies are checked as well, so the `redshiftTools` check fails. Steps to reproduce:

+ `R CMD BUILD .`
+ `R CMD CHECK <tarball>`

**Comment by durch** [comment]
@russellpierce finally figured it out, tough I do not understand it. `R CMD build` pulls in `zapieR` source, and thats whats been erroring out, [this](https://github.com/zapier/statistics/pull/351) PR handles that.

Any ideas on how it knows where to pull the sources from?

**Comment by russellpierce** [comment]
I mean it knows zapieR is a dependency, so it is free to check .libLocs() to find zapieR if it wants to byte encode (kinda a compile step R added a few years ago that I never paid much attention to). The solution being to change functions inside zapieR totally unrelated to redshift tools to be explicit makes no sense (much less making only some of the calls explicit). Nor does running zapieR tests for redshiftTools. I also tried the command line build process (with the non-updated zapieR) and hit no errors. I wonder if your build directory got messed up with some non-commited change. Oh well, doesn't matter much now. If you hit this type of error again, please let's pair. For this kind of thing being able to see/work with the environment that is funky is invaluable.

**Pull request merged**